### PR TITLE
Add Jenkinsfile to the repo

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This change adds Jenkinsfile for jenkins.io. I have not tested it (not sure how to reliably do it).

CC @rtyler and @reviewbybees

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        stage("build") {
+            steps {
+                sh 'mvn clean verify install -Dmaven.test.failure.ignore=true'
+            }
+        }
+    }
+    
+    post {
+        always {
+            junit '*/target/surefire-reports/*.xml'
```
Comment (oleg-nenashev) [inline Jenkinsfile]: I would like to also publish FindBugs, but seems the publisher still does not support symbols

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    
+    agent {
+        label "java"
+    }
+    
+    // TODO: Copy-pasted from the Pipeline Model Definition example
+    // Make sure we have GIT_COMMITTER_NAME and GIT_COMMITTER_EMAIL set due to machine weirdness.
+    environment {
+        GIT_COMMITTER_NAME = "jenkins"
+        GIT_COMMITTER_EMAIL = "jenkins@jenkins.io"
```
Comment (jglick) [inline Jenkinsfile]: What here is committing to Git?

File: Jenkinsfile (hunk 1 above)
Comment (jglick) [inline Jenkinsfile]: `step`

**Comment by oleg-nenashev** [comment]
Yes, should be buildPlugin() . Or buildModule() just to have a planB for
the future

2017-02-27 16:58 GMT+01:00 Jesse Glick <notifications@github.com>:

**Comment by oleg-nenashev** [comment]
@reviewbybees So I have created `buildModule()` in https://github.com/jenkins-infra/pipeline-library/pull/6 . Now this PR is de-facto a one-liner

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,3 @@
+
+// Builds a module using https://github.com/jenkins-infra/pipeline-library
+buildModule(platforms: ['linux'], allowEmptyTestResults: true)
```
Comment (jglick) [inline Jenkinsfile]: I would be more comfortable deleting

```xml
<disabledTestInjection>true</disabledTestInjection>
```

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,3 @@
+
+// Builds a module using https://github.com/jenkins-infra/pipeline-library
```
Comment (jglick) [inline Jenkinsfile]: I think you can actually test the upstream PR prior to it being merged, using something like

```groovy
@Library('github.com/oleg-nenashev/pipeline-library@7b106252928c000a81734f64eaa84f822a680134') _
```

Of course this only matters if [organization scanning](https://ci.jenkins.io/job/Plugins/computation/) is fixed while both PRs are open.

Comment (oleg-nenashev) [inline Jenkinsfile]: For local testing I use https://github.com/jenkinsci/filesystem_scm-plugin/pull/2.
For upstream PR testing I would like to get it hosted first :)

**Comment by oleg-nenashev** [comment]
@jglick migrated to `buildPlugin()` with the stock version of the lib

File: pom.xml [hunk 5]
```diff
@@ -37,7 +37,7 @@
         <artifactId>maven-hpi-plugin</artifactId>
         <extensions>true</extensions>
         <configuration>
-          <disabledTestInjection>true</disabledTestInjection>
+          <disabledTestInjection>false</disabledTestInjection>
```
Comment (jglick) [inline pom.xml]: Or just simply delete the whole plugin section, since you are not overriding anything any more.

**Comment by oleg-nenashev** [comment]
@reviewbybees done.
Merging as a maintainer

**Pull request merged**